tools/pythonpkg/src/vector_conversion.cpp

```diff
@@ -294,7 +294,7 @@ void VectorConversion::NumpyToDuckDB(PandasColumnBindData &bind_data, py::array
 		//! We have determined the underlying logical type of this object column
 		// Get the source pointer of the numpy array
 		auto src_ptr = (PyObject **)numpy_col.data();
-		if (out.GetType().id() != LogicalTypeId::VARCHAR) {
+		if (!bind_data.all_strings) {
 			return ScanPandasObjectColumn(bind_data, src_ptr, count, offset, out);
```

Comment:
I don't quite follow why the different code path is required here - we also do object conversion below if the object is not of a built-in string type (`!PyUnicode_CheckExact(val)`). Can we instead remove `ScanPandasObjectColumn` entirely and replace the generic object conversion below with the correct object conversion code?

Reply:
Previously it got here:
```c++
			if (!PyUnicode_CheckExact(val)) {
				out_mask.SetInvalid(row);
				continue;
			}
```
and that's why it got turned into None
The only thing I've done is made it so that objects that should convert to VARCHAR but aren't of type `builtin.str` now don't take this fast path because evidently that can't handle it correctly

Reply:
Ah now I get it, the problem is likely that `numpy.str_` inherits from `builtins.str` but doesn't behave the same.
We just need a `py::type::of(val) == py::module_::import("builtins").attr("str"))` instead of the `py::isinstance` stuff

Reply:
That works, I have had to move the gil lock outside of the body because of the need to import the builtins module, which isn't ideal..

Maybe it's better to just not import `builtins` lazily, since we need it for exact type checks like this one
and then we don't need the GIL here, since we can be sure that builtins is already loaded

Reply:
Isn't calling `PyUnicode_CheckExact` sufficient as a string-type check, or does the built-in module add any additional functionality?